src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/resource/CalculationControllerTest.kt

```diff
@@ -464,5 +465,63 @@ class CalculationControllerTest {
       .isEqualTo(expected)
   }
 
+  @Test
+  fun `Test GET of offender key dates for a booking id successfully`() {
+    val calcRequestId = 5636121L
+    val bookingId = 123456L
+    val expected = OffenderReleaseDates(
+      bookingId,
+      calcRequestId,
+      LocalDateTime.of(2024, 2, 29, 10, 30),
+      "14 day check",
+      CalculationSource.CRDS,
+      listOf(
+        DetailedDate(
+          ReleaseDateType.HDCED,
+          ReleaseDateType.HDCED.description,
+          LocalDate.of(2024, 1, 1),
+          emptyList(),
+        ),
+      ),
+    )
+
+    whenever(offenderKeyDatesService.getKeyDatesByCalcId(any())).thenReturn(expected)
+
+    val result = mvc.perform(get("/calculation/release-dates/$calcRequestId").accept(APPLICATION_JSON))
+      .andExpect(status().isOk)
+      .andExpect(content().contentType(APPLICATION_JSON))
+      .andReturn()
+
+    assertThat(mapper.readValue(result.response.contentAsString, OffenderReleaseDates::class.java))
+      .isEqualTo(expected)
+  }
+
+  @Test
+  fun `Test GET of release Dates when there is a problem getting the release dates using CalcReleaseId`() {
+    val calcRequestId = 5636121L
+    val errorMessage = "There isn't one"
+
+    whenever(
+      offenderKeyDatesService.getKeyDatesByCalcId(any()),
+    ).then {
+      throw CrdWebException(errorMessage, HttpStatus.NOT_FOUND)
+    }
+
+    val result = mvc.perform(get("/calculation/release-dates/$calcRequestId").accept(APPLICATION_JSON))
+      .andExpect(status().isNotFound)
+      .andExpect(content().contentType(APPLICATION_JSON))
+      .andReturn()
+
+    assertThat(mapper.readValue(result.response.contentAsString, ErrorResponse::class.java))
+      .isEqualTo(
+        ErrorResponse(
+          HttpStatus.NOT_FOUND,
+          null,
+          errorMessage,
+          errorMessage,
+        ),
+      )
+  }
+
```

Comment:
missing test for offender key dates returned an error (Left)

Reply:
Add 3 new cases as discussed in OffenderKeyDatesServiceTest.kt